app/src/main/java/com/example/sharewhatyoucanproject/rdashboard/rdetailedPosts/RDetailsFragment.kt

```diff
@@ -0,0 +1,52 @@
+package com.example.sharewhatyoucanproject.rdashboard.rdetailedPosts
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModelProvider
+import androidx.navigation.fragment.navArgs
+import com.bumptech.glide.Glide
+import com.example.sharewhatyoucanproject.databinding.FragmentRDetailsBinding
+import com.example.sharewhatyoucanproject.models.PostModel
+import com.example.sharewhatyoucanproject.rposts.RPostViewModel
+import com.example.sharewhatyoucanproject.rposts.RPostViewModelFactory
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+
+class RDetailsFragment : Fragment() {
+
+    private var _binding: FragmentRDetailsBinding? = null
```

Comment:
make sure to null out the binding in `onDestroyView`

Reply:
done